### ISSUE TITLE
ENH: Gracefully handle empty intervals in difference (or other operators)

### ISSUE THREAD

**Issue by damien-sornette** [staircase-dev/piso#56]
#### Problem
Currently, passing an empty IntervalArray or IntervalSet to the difference operator will cause an error to be thrown. It's not so much a problem when you just make a difference between two elements, a simple if will protect you. However, it becomes a bit tedious to handle with 3 elements where the second and third can sometimes be empty. The number of possible cases becomes exponentially higher when you add more sets.

Of course you can add the arguments in an array and the iterate over them to "accumulate" the difference, but it seems overcomplicated for something that could be handled gracefully by the lib

Unless there is a reason not to ?

#### Solution
When an empty argument is passed, it is ignored, and the reference (leftmost argument) interval is returned. And if the left most argument is empty, then return it.

**Comment by venaturum** [comment]
Hi @damien-sornette , thanks for opening this issue.  I'll look to implement and publish within the next month unless I hit an obstacle.  Cheers.

**Comment by damien-sornette** [comment]
Great, thanks ! I'll keep an eye on this issue then :)

**Comment by venaturum** [comment]
Hi @damien-sornette,

Are you able to provide a reproducible example.  The following works ok for me so maybe I'm not imagining the situation properly

```python
import pandas as pd
import piso
piso.register_accessors()

arr1 = pd.arrays.IntervalArray.from_tuples(
    [(0, 4), (2, 5), (3, 6), (7, 8), (8, 9), (10, 12)],
)

arr2 = pd.arrays.IntervalArray.from_tuples([])

piso.difference(arr1, arr2)
arr1.piso.difference(arr2)
```

Are you using the latest version of piso?

Cheers.

**Comment by damien-sornette** [comment]
I'm currently using piso 0.9.0, it seems to be the last version available on pip.

Your code work for me, but this example doesn't :

```python
import pandas as pd
from pandas.core.arrays import IntervalArray
from piso import difference

if __name__ == '__main__':
    an_interval = pd.Interval(pd.Timestamp('2017-01-01T12'), pd.Timestamp('2018-01-01T12'), closed="left")
    difference(IntervalArray([an_interval]), IntervalArray([]))
```

I built this case to look as close as possible to my real code.

The main difference I see is that I'm using a different way to build my Interval, and it's a time interval, not an int interval.

Let me know if you need more infos !

**Comment by venaturum** [comment]
Closed with #58 and release of piso 1.0.0
